deploy-board/deploy_board/static/js/components/capacitycomponents.js

```diff
@@ -214,30 +257,95 @@ Vue.component("static-capacity-config", {
+                <span class="input-group-addon">Min Size</span>
+                <input name="minSize" class="form-control" type="number" min="0" required
+                    :value="minSize" @input="onMinSizeChange($event.target.value)" @keydown.enter.prevent="" >
+            </div>
+        </div>
+        <div :class="inputBootstrapClass">
+            <div class="input-group">
+                <span class="input-group-addon">Max Size</span>
+                <input name="maxSize" class="form-control" type="number" min="0" required
+                    :value="maxSize" @input="onMaxSizeChange($event.target.value)" @keydown.enter.prevent="">
+            </div>
+        </div>
+    </div>
+    <form-warning v-show="showSizeWarning" :alert-text="sizeWarning"></form-warning>
+    <form-error v-show="showSizeError" :alert-text="sizeError"></form-error>
+    </div>`,
+    props: {
+        labelBootstrapClass: {
+            type: String,
+            default: 'col-xs-4',
+        },
+        inputBootstrapClass: {
+            type: String,
+            default: 'col-xs-2',
+        },
+        labelText: {
+            type: String,
+            default: 'Number of hosts for this service',
         },
-        updateMaxSize: function (value) {
-            this.$emit('maxchange', value)
+        labelTitle: {
+            type: String,
+            default: 'Capacity'
+        },
+        originalMinSize: Number,
+        originalMaxSize: Number,
+        remainingCapacity: Number,
+        placements: Object,
+    },
+    data: function() {
+        return {
+            minSize: this.originalMinSize,
+            maxSize: this.originalMaxSize,
+            showSizeError: false,
+            showSizeWarning: false,
+            sizeError: '',
+            sizeWarning: '',
         }
+    },
+    methods: {
+        onMinSizeChange: function (value) {
+            this.minSize = Number(value);
+            if (this.maxSize < this.minSize) {
+                this.maxSize = this.minSize;
+            }
+            this.validateSize();
+            this.$emit('min-change', this.minSize);
+        },
+        onMaxSizeChange: function (value) {
+            this.maxSize = Number(value);
+            if (this.maxSize < this.minSize) {
+                this.minSize = this.maxSize;
+            }
+            this.validateSize();
+            this.$emit('max-change', this.maxSize);
+        },
+        validateSize: function() {
+            const minIncrease = this.minSize - this.originalMinSize;
+            const maxIncrease = this.maxSize - this.originalMaxSize;
+
+            if (minIncrease >= this.remainingCapacity) {
+                this.sizeError = getCapacityAlertMessage(false, this.remainingCapacity, this.placements, minIncrease);
+                this.showSizeError = true;
+            } else {
+                this.showSizeError = false;
+            }
+
+            if(!this.showSizeError && maxIncrease >= this.remainingCapacity){
```

Comment:
can't both conditions need warnings? it seems they are independent, e.g. if both are 1 million, we should display both messages

Reply:
Or do both have the same display but different severity? I will check

Reply:
The idea is to show the worse problem. When the scenario you mentioned happens, an error shows first. When the error (min size) is corrected, the warning shows util the max size is corrected.

Reply:
it would be better to show all problems upfront if possible, otherwise the user is waiting on one error to see the next error, sort of like when we have a compile failure, then find more compile failures are waiting after. we can avoid that by showing all problems

Reply:
This has some subtle differences. A user has to over-resize once, sees and ignores the first alert and then over-resizes again to trigger the second alert (hiding if it's a warning or overriding if it's an error). I will however remove the check to show both.